### PR TITLE
Change _convert_custom_fields_to_struct function...

### PR DESCRIPTION
**Pull request by kmtracey**
that is called for each row of the uploaded file into _create_field_map method of _Importer object, called only once per import.

While in the area move a couple of other functions only called from _Importer to being methods on _Importer.

## Summary

Noticed inefficiency when doing initial work on adding data validation to case importer.

## Feature Flag

None

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [ ] I have confidence that this PR will not introduce a regression for the reasons below

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

**Comment by orangejenny** [comment]
@calellowitz Can you review this one? I realize you're offline at the moment, but this isn't urgent, reviewing within the next week or two is fine.

File: corehq/apps/case_importer/do_import.py [hunk 1]
```diff
@@ -195,6 +196,92 @@ def submit_case_blocks(self, caseblocks):
...
+
+            fields_to_update[update_field_name] = update_value
+
+        return fields_to_update
+
+    def _create_field_map(self):
+        config = self.config
+        excel_fields = config.excel_fields
+        case_fields = config.case_fields
+        custom_fields = config.custom_fields
```
Comment (calellowitz) [inline corehq/apps/case_importer/do_import.py:261]: Are these variables necessary? The method uses attributes of the `config` directly below a number of times.

**Review by calellowitz — COMMENTED** [review]
The main change here looks excellent. That was a great find. 

The movement of some of the helper functions to methods created a little bit of inconsistency, there are still some helper functions at the top level, and some as methods. I have a slight preference for leaving those the way they are, rather than moving them without changing the substance, but could be convinced if there are clear advantages to methods here. If so, it would be nice to move the remaining ones.

File: corehq/apps/case_importer/do_import.py (hunk 1 above)
Comment (kmtracey) [reply corehq/apps/case_importer/do_import.py:261]: They don't look to be strictly necessary, they were there in the original function and in the interest of making minimal changes were left unchanged. It isn't clear to me why these exist (looks to be only 2 uses of 2 of them and 1 of the first) while the `config.search_` vars are used directly below.

**Comment by kmtracey** [comment]
re: helper function/method inconsistency...it looks like `_log_case_lookup` and `_convert_field_value` are the remaining helper functions not moved into methods? The ones moved were the ones that took `config` as a parameter and these did not, so those were all converted to methods accessing `self.config`. `log_case_lookup` is used twice and is handed `self.domain`, that could be converted and access `self.domain` instead of having `domain` passed in. `convert_field_value` doesn't use anything stored on the object so could be converted but it would be a method that doesn't use `self`. I have no strong opinions on which is "better"...

File: corehq/apps/case_importer/do_import.py (hunk 1 above)
Comment (calellowitz) [reply corehq/apps/case_importer/do_import.py:261]: that makes sense, good to leave them here then

**Comment by calellowitz** [comment]
> re: helper function/method inconsistency...it looks like `_log_case_lookup` and `_convert_field_value` are the remaining helper functions not moved into methods? The ones moved were the ones that took `config` as a parameter and these did not, so those were all converted to methods accessing `self.config`. `log_case_lookup` is used twice and is handed `self.domain`, that could be converted and access `self.domain` instead of having `domain` passed in. `convert_field_value` doesn't use anything stored on the object so could be converted but it would be a method that doesn't use `self`. I have no strong opinions on which is "better"...

That all makes sense, I think this is good to go

**Review by calellowitz — APPROVED** [review]

**Comment by orangejenny** [comment]
Deployed to staging and ran a basic smoke test of case imports. This is a low enough risk change that I don't think formal QA is necessary. Merging!

**Pull request merged**